server/src-lib/Hasura/GraphQL/RemoteServer.hs

```diff
@@ -217,7 +220,9 @@ instance J.FromJSON (FromIntrospection G.UnionTypeDefinition) where
     kind  <- o .: "kind"
     name  <- o .:  "name"
     desc  <- o .:? "description"
-    possibleTypes <- o .: "possibleTypes"
+    -- TODO: maybe instead of using `RemoteSchemaInputValueDefinition` here
+    -- we can generalize it, like in other places?
+    possibleTypes :: [FromIntrospection (G.ObjectTypeDefinition RemoteSchemaInputValueDefinition)] <- o .: "possibleTypes"
```

Comment:
I'm not sure how to do this, the other instances are 

```haskell
instance (J.FromJSON (FromIntrospection a)) => J.FromJSON (FromIntrospection (G.InterfaceTypeDefinition [G.Name] a)) where
```

but the `G.UnionTypeDefinition` doesn't depend on `a`, but to get the `possibleTypes` we need to parse the object definitions and then get the names out of those objects.